llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -803,6 +805,13 @@ class VPRecipeBase : public ilist_node_with_parent<VPRecipeBase, VPBasicBlock>,
   /// Returns the debug location of the recipe.
   DebugLoc getDebugLoc() const { return DL; }
 
+  /// Returns the IR flags for the recipe.
+  virtual const VPRecipeIRFlags *getIRFlags() const { return nullptr; }
```

Comment:
Would it be possible to still retain the option to cast/dyn_cast to VPRecipeIRFlags by implementing VPRecipeIRFlags::classof for VPRecipeBase instead?

Reply:
I'll see... :thinking:  I'm not convinced that would be a better solution. It makes it required for recipes to inherit from `VPRecipeIRFlags` (so `VPRecipeBase*` is castable to `VPRecipeIRFlags*`), and add the additional maintenance keeping the list of classes in the `classof` -- without a `getIRFlags()` on the base). 

Given there's only one place that did a VPValue -> IRFlags cast before (the rest easily replaced with `.getIRFlags()`), it does not seem like a critical feature.